tests/feature_tests/sizeof.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+
+int main() {
```

Comment:
Thank you for adding this very comprehensive test! Just one more thing I thought of - could you add a couple lines to make sure `sizeof` works correctly on a typedefed type, and also that a struct type declared inside a sizeof is visible outside the sizeof function? I'm thinking something like:
```
typedef int T;
if(sizeof(T) != 4) return 1;
  
T a;
if(sizeof(a) != 4) return 2;
  
sizeof(struct S {int a;});
struct S s;
```